packages/gatsby-theme-api-docs/src/components/type/properties/rows/name-type.js

```diff
@@ -13,17 +12,11 @@ const NameType = ({
   parentDiscriminator,
   discriminatorValue,
 }) => {
-  const typeLocations = useTypeLocations();
-  const dataForTypeToLinkTo =
-    property.type === 'array' && property.items
-      ? { type: property.items.type, displayPrefix: 'Array of ' }
-      : { type: property.type, displayPrefix: '' };
-
-  const type = renderTypeAsLink(
+  const typeToRender = generateTypeToRender({
+    typeLocations: useTypeLocations(),
```

Comment:
Please keep usage of react hooks within the body function, otherwise it's confusing and might lead to unnecessary issues (for example render a hook within a condition).

```suggestion
  const typeLocations = useTypeLocations();
  const typeToRender = generateTypeToRender({
    typeLocations,
```